src/scippneutron/file_loading/nxobject.py

```diff
@@ -1,20 +1,55 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock
+import numpy as np
+import scipp as sc
 from enum import Enum, auto
 import functools
+from typing import List, Union, Tuple, NoReturn, Any, Dict
 
 from ._nexus import LoadFromNexus
-from ..file_loading._hdf5_nexus import LoadFromHdf5
-from ._common import Group, Dataset
+from ._hdf5_nexus import LoadFromHdf5
+from ._common import Group, Dataset, MissingAttribute
+
+ScippIndex = Tuple[str, Union[int, slice]]
+NXobjectIndex = Union[str, ScippIndex]
 
 
 class NX_class(Enum):
-    NXroot = auto()
+    NXdata = auto()
+    NXdetector = auto()
     NXentry = auto()
+    NXevent_data = auto()
     NXlog = auto()
     NXmonitor = auto()
-    NXevent_data = auto()
+    NXroot = auto()
+
+
+class Attrs:
+    """HDF5 attributes.
+    """
+    def __init__(self,
+                 node: Union[Dataset, Group],
+                 loader: LoadFromNexus = LoadFromHdf5()):
+        self._node = node
+        self._loader = loader
+
+    def __contains__(self, name: str) -> bool:
+        try:
+            _ = self[name]
+            return True
+        except MissingAttribute:
+            return False
+
+    def __getitem__(self, name: str) -> Any:
+        attr = self._loader.get_attribute(self._node, name)
+        # Is this check for string attributes sufficient? Is there a better way?
+        if isinstance(attr, str) or isinstance(attr, bytes):
```

Comment:
```suggestion
        if isinstance(attr, (str, bytes)):
```

I think this is sufficient. There are no other string types to my knowledge. (Unless you count OPAQUE datatypes which we cannot decode but could unwrap and store as `bytes`. See https://docs.h5py.org/en/stable/strings.html#how-to-store-raw-binary-data)